iam-login-service/src/main/java/it/infn/mw/iam/api/scim/provisioning/ScimUserProvisioning.java

```diff
@@ -542,8 +556,75 @@ private void handleSpecificUpdateType(IamAccount account, AccountUpdater u) {
   public void update(final String id, final List<ScimPatchOperation<ScimUser>> operations) {
 
     IamAccount account = accountRepository.findByUuid(id).orElseThrow(() -> noUserMappedToId(id));
+    Optional<IamAccount> currentUserAccount = getCurrentUserAccount();  
+    if (!currentUserAccount.isPresent() || isAdmin(currentUserAccount.get())) {
+      operations.forEach(op -> executePatchOperation(account, op));
+    } else {
+      operations.forEach(op -> executePatchOperationByUser(account, op));
+    }
+  }
+
+  private boolean isAdmin(IamAccount userAccount) {
+    return userAccount.getAuthorities().stream()
+        .anyMatch(IamAuthority::isAdminAuthority);
+  }
+
+  private void executePatchOperationByUser(IamAccount account, ScimPatchOperation<ScimUser> op) {
+    Set<UpdaterType> enabledUpdaters = getEnabledUpdaters();
+
+    List<AccountUpdater> updaters = updatersFactory.getUpdatersForPatchOperation(account, op);
+
+    for (AccountUpdater updater : updaters) {
+      if (!enabledUpdaters.contains(updater.getType())) {
+        throw new ScimPatchOperationNotSupported(updater.getType().getDescription() + " not supported");
+      }
+    }
+
+    List<AccountUpdater> updatesToPublish = updaters.stream()
+        .filter(AccountUpdater::update)
+        .toList();
 
-    operations.forEach(op -> executePatchOperation(account, op));
+    if (!updatesToPublish.isEmpty()) {
+      account.touch();
+      accountRepository.save(account);
+      updatesToPublish.forEach(u -> u.publishUpdateEvent(this, eventPublisher));
+    }
+  }
+
+  public Set<UpdaterType> getEnabledUpdaters() {
+    EnumSet<UpdaterType> enabledUpdaters = EnumSet.noneOf(UpdaterType.class);
+
+    enabledUpdaters.addAll(ACCOUNT_LINKING_UPDATERS);
+
+    iamProperties.getUserProfile().getEditableFields().forEach(e -> {
+      if (EditableFields.NAME.equals(e)) {
+        enabledUpdaters.add(ACCOUNT_REPLACE_GIVEN_NAME);
+      } else if (EditableFields.SURNAME.equals(e)) {
+        enabledUpdaters.add(ACCOUNT_REPLACE_FAMILY_NAME);
+      } else if (EditableFields.PICTURE.equals(e)) {
+        enabledUpdaters.add(ACCOUNT_REPLACE_PICTURE);
+        enabledUpdaters.add(ACCOUNT_REMOVE_PICTURE);
+      } else if (EditableFields.EMAIL.equals(e)) {
```

Comment:
This has not been introduced by you but I'm thinking about giving the opportunity to the user to change the email address: we should send an email to verify it! Probably we should avoid letting users to change their email (by default) and we should work on introducing a replace mechanism that verifies the new email. Not so easy.

Reply:
by default email is included in the editable fields: I'd remove it honestly https://github.com/indigo-iam/iam/blob/develop/iam-login-service/src/main/resources/application.yml#L166